shared/dev/dumb-sheet/render.native.js

```diff
@@ -64,63 +68,95 @@ class DumbSheetRender extends Component<void, Props, any> {
     return this.props.autoIncrement ? this.renderIncrement() : this.renderSingle()
   }
 
-  _getComponents (filter: ?string) {
-    const components = []
-    const componentsOnly = []
-    const parentPropsOnly = []
+  _getTotal (filter: ?string) {
+    let total = 0
+    Object.keys(dumbComponentMap).forEach(key => {
+      if (filter && key.toLowerCase().indexOf(filter) === -1) {
+        return
+      }
+
+      const map = dumbComponentMap[key]
+      total += Object.keys(map.mocks).length
+    })
+
+    return total
+  }
+
+  _getComponent (filter: ?string, renderIdx: number) {
+    let component = null
+    let parentProps = {}
+    let mockStore = null
 
+    let currentIdx = 0
     Object.keys(dumbComponentMap).forEach(key => {
       if (filter && key.toLowerCase().indexOf(filter) === -1) {
         return
       }
 
       const map = dumbComponentMap[key]
       const Component = map.component
+
       Object.keys(map.mocks).forEach((mockKey, idx) => {
-        const mock = {...map.mocks[mockKey]}
-        const parentProps = mock.parentProps
-        mock.parentProps = undefined
-
-        components.push(
-          <Box key={mockKey} style={styleBox}>
-            <Text type='BodySmall'>{key}: {mockKey}</Text>
-            <Box {...parentProps}>
-              <Component key={mockKey} {...mock} />
-            </Box>
-          </Box>
-        )
-        componentsOnly.push(<Component key={mockKey} {...mock} />)
-        parentPropsOnly.push(parentProps)
+        if (renderIdx === currentIdx) {
+          const mock = {...map.mocks[mockKey]}
+          parentProps = mock.parentProps || {}
+          mockStore = mock.mockStore
+          mock.parentProps = undefined
+          mock.mockStore = undefined
+          component = <Component key={mockKey} {...mock} />
+        }
+        ++currentIdx
       })
     })
 
     return {
-      components,
-      componentsOnly,
-      parentPropsOnly,
+      component,
+      parentProps,
+      mockStore,
     }
   }
 
   renderIncrement () {
     if (this.state.testIndex === -1) {
       return <Text type='Body'>DONE TESTING</Text>
     }
-    const {componentsOnly} = this._getComponents()
-    const sub = componentsOnly.slice(this.state.testIndex, this.state.testIndex + 1)
-    console.log('test render idx', this.state.testIndex, sub)
-    return <Box>{sub}</Box>
+    const {component, mockStore} = this._getComponent(null, this.state.testIndex)
+
+    this._updateMockStore(mockStore)
+    console.log('test render idx', this.state.testIndex, component)
+    return <Box>{this._makeStoreWrapper(component)}</Box>
+  }
+
+  _updateMockStore (mockStore) {
+    if (mockStore) {
```

Comment:
Do what desktop does. On RN you'll get a setState message but its likely not fixing this as this is dev-only and likely a rabbit hole